pymess/backend/dialer/daktela.py

```diff
@@ -0,0 +1,130 @@
+
+    def _update_dialer_states(self, messages):
+        """
+        Method uses Daktela API to get info about autodialer call status
+        :param messages: list of dialer messages to update
+        """
+        for message in messages:
+            name = message.extra_data['name']
+            client_url = self._get_dialer_api_url(name)
+            response = generate_session(slug=self.SESSION_SLUG, related_objects=(message,)).get(client_url)
+            resp_json = response.json()
+
+            message.extra_data.update({
+                'name': name,
+                'daktela_action': resp_json['result']['action'],
+                'daktela_statuses': resp_json['result']['statuses'],
+            })
+            resp_message_state = resp_json['result']['statuses'][0]['name'] if len(
+                resp_json['result']['statuses']) else resp_json['result']['action']
+            message_state = self.STATES_MAPPING[resp_message_state]
+            message_error = resp_json['error'] if len(resp_json['error']) else None
+
+            try:
+                self.update_message(
+                    message,
+                    state=message_state,
+                    error=message_error,
+                    extra_data=message.extra_data,
+                )
+            except Exception as ex:
+                self.update_message(message, state=DialerMessage.STATE.ERROR, error=force_text(ex))
+                raise ex
+
+    def publish_message(self, message):
+        """
+        Method uses Daktela API for sending dialer message
+        :param message: dialer message
+        """
+        client_url = self._get_dialer_api_url()
+        try:
+            payload = {
+                'queue': settings.DIALER_API_QUEUE,
+                'number': message.recipient,
+                'customFields': {
+                    'mall_pay_text': [
+                        message.content,
+                    ],
+                    'ttsprocessed': [
+                        0,
+                    ],
+                },
+                'action': 5,
+            }
+
+            response = generate_session(slug=self.SESSION_SLUG, related_objects=(message,)).post(
+                client_url,
+                json=payload,
+            )
+            resp_json = response.json()
+
+            message.extra_data.update({
+                'name': resp_json['result']['name'],
+                'daktela_action': resp_json['result']['action'],
+                'daktela_statuses': resp_json['result']['statuses'],
+            })
+
+            self.update_message(
+                message,
+                state=DialerMessage.STATE.READY,
+                error=resp_json['error'] if len(resp_json['error']) else None,
+                sent_at=tz.now(),
+                extra_data=message.extra_data,
+            )
+        except Exception as ex:
+            self.update_message(message, state=DialerMessage.STATE.ERROR, error=force_text(ex))
```

Comment:
když se to re-raisuje nahoře, tak ať je to i tady

Reply:
OK